daemon/archive_tarcopyoptions_unix.go

```diff
@@ -25,7 +25,7 @@ func (daemon *Daemon) tarCopyOptions(ctr *container.Container, noOverwriteDirNon
 	}
 
 	return &archive.TarOptions{
-		NoOverwriteDirNonDir: noOverwriteDirNonDir,
+		NoOverwriteDirNonDir: !allowOverwriteDirWithFile,
```

Comment:
FWIW; considering we should make pkg/archive follow;

- Add a `AllowOverWriteDirWithFile` option
- Deprecate `NoOverwriteDirNonDir` and make it the default


I opened a quick PR where I removed the option, and nothing looks to be breaking, but maybe there's some specific use-case that requires it (unpacking an archive to a directory or something); maybe someone knows?

- https://github.com/moby/moby/pull/50401